dropshot/src/versioning.rs

```diff
@@ -148,7 +153,21 @@ impl ClientSpecifiesVersionInHeader {
         name: HeaderName,
         max_version: Version,
     ) -> ClientSpecifiesVersionInHeader {
-        ClientSpecifiesVersionInHeader { name, max_version }
+        ClientSpecifiesVersionInHeader { name, max_version, on_missing: None }
+    }
+
+    /// If the header is missing, use the provided version instead.
```

Comment:
Can we update this comment to dissuade people from using this if they control the clients (and maybe explicitly say not for Oxide internal services?)

Reply:
Done, and bolded.